jscomp/others/js_date.ml

```diff
@@ -130,6 +130,7 @@ external toJSON : t -> string = "" [@@bs.send]
 [@@ocaml.deprecated "This method is unsafe. It will be changed to return option in a future release. Please use toJSONUnsafe instead."]
 external toJSONUnsafe : t -> string = "toJSON" [@@bs.send]
 external toLocaleDateString : t -> string = "" [@@bs.send] (* TODO: has overloads with somewhat poor browser support *)
+external toLocaleDateStringWithLocale : t -> locale:string -> ?options:'a -> unit -> string = "toLocaleDateString" [@@bs.send] (* TODO: has overloads with somewhat poor browser support *)
```

Comment:
the type of `options` seems to be too loose to introduce some issues?

Reply:
Thanks for the feedback. Added Js.t type constraint.